streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStandbyTasks.java

```diff
@@ -24,4 +24,14 @@
         super(logContext, "standby task");
     }
 
+    @Override
+    int commit() {
+        final int committed = super.commit();
+        // TODO: this contortion would not be necessary if we got rid of the two-step
+        // task.commitNeeded and task.commit and instead just had task.commitIfNeeded. Currently
+        // we only call commit if commitNeeded is true, which means that we need a way to indicate
+        // that we are eligible for updating the offset limit outside of commit.
+        running.forEach((id, task) -> task.allowUpdateOfOffsetLimit());
```

Comment:
Seems we should add `AssignedStandbyTasksTest.java` to unit test the new `commit()` behavior.

Reply:
We could add a test here, but it is super implementation specific. This is not really the best home for the code, just the most convenient at the moment. All we would test, I think, is that we called allowUpdateOfOffsetLimit? The integration test covers this in a more comprehensive way and doesn't break when we move this code. What do you think?

Reply:
> All we would test, I think, is that we called allowUpdateOfOffsetLimit?

Yes, the test should be `shouldUpdateOffsetsLimtOnCommit()`.

> The integration test covers this in a more comprehensive way and doesn't break when we move this code.

"and _doesn't_ break" -- I guess it should break? The main purpose of the unit test is, that is would be hard to debug the integration test and figure out the root cause why it fails.

Reply:
I added back the StandbyTask tests that will cover this with a much more narrow test - specifically to cover when to update the offset limit. If this gets moved into StandbyTask that test should not fail, provided it was done right. If it's broken in any way that test will flag it and it should be easy to look at the test failure and the changed code to determine the cause.

Do you think StandbyTaskTest sufficiently addresses your concern?